typed/compare.go

```diff
@@ -224,11 +232,14 @@ func (w *compareWalker) visitListItems(t *schema.List, lhs, rhs value.List) (err
 			// this element.
 			continue
 		}
-		// Ignore repeated occurences of `pe`.
-		if _, found := lValues.Get(pe); found {
-			continue
+
+		if v, found := lValues.Get(pe); found {
+			list := v.Unstructured().([]value.Value)
```

Comment:
Would it be preferable to avoid unstructured conversion for the comparison? This kinda makes me sad

Reply:
Yeah I dont understand how this line works. Shouldn't `Unstructured` return a []interface{}?

Reply:
Yeah, it's because I'm abusing the PathElementValueMap here. I don't insert a valid value in there, I cast a `[]value.Value` into a value and then retrieve it, and it just works. I don't know if we want to introduce generics code in here just yet, so I can't really make the map take whatever type of parameter type I want.